### PR TITLE
Copy persistent values in `Link.copyparams`

### LINKED ISSUE #4903: Should `copyparams` copy persistent values as well?
Currently, `{Link, Chain}.copyparams` does not care persistent values and it only copies parameters. Due to this behavior, for example, *every* model using batch normalization cannot be correctly copied using this method.

See #4902 for an example of errors due to this problem.

Do we have situations where we want to copy parameter values but we don't want to copy persistent values? I propose to change the behavior of `copyparams` to also copy persistent values.

### PR DESCRIPTION
**Pull request by beam2d**
Fixes #4903. This change makes `Link.copyparams` also copies the persistent values, notably the `avg_mean`, `avg_var`, and `N` of `links.BatchNormalization`.

Note that this change is backward incompatible since the default behavior is changed. The original behavior is confusing, and the new behavior should be beneficial in most cases (including existing code). The original behavior can be reproduced by passing `copy_persistent=False`.

### CODE REVIEW

**Comment by beam2d** [comment]
This PR also includes a utility function `cuda.copyto`, which simplifies the implementation.

File: chainer/link.py [hunk 1]
```diff
@@ -508,21 +508,38 @@ def children(self):
...
+            copy_persistent (bool): If ``True``, persistent values are also
+                copied. ``True`` by default.
 
         """
         src = link.__dict__
         dst = self.__dict__
         for name in self._params:
             dst[name].copydata(src[name])
+        if copy_persistent:
+            array_types = numpy.ndarray, cuda.ndarray
```
Comment (kmaehashi) [inline chainer/link.py]: Use `chainer.get_array_types()`.

**Comment by kmaehashi** [comment]
During the review I found another issue regarding iDeep: #5009. (Let's fix it after merging this PR)

File: chainer/link.py (hunk 1 above)
Comment (beam2d) [reply chainer/link.py]: fixed

**Comment by beam2d** [comment]
I'll fix for ideep after merging #5009.

**Comment by beam2d** [comment]
I fixed for the comment. The iDeep issue is left in this PR.

File: tests/chainer_tests/test_link.py [hunk 2]
```diff
@@ -397,6 +401,22 @@ def test_copyparams(self):
...
+        l, gs = self._setup_test_copyparams()
+        self.link.copyparams(l)
+        self._check_copyparams(l, gs)
+        numpy.testing.assert_array_equal(self.link.p, l.p)
+
+    def test_copyparams_no_copy_persistent(self):
+        orig_p = self.link.p.copy()
+
+        l, gs = self._setup_test_copyparams()
+        numpy.testing.assert_array_equal(False, orig_p == l)
```
Comment (kmaehashi) [inline tests/chainer_tests/test_link.py]: `l` -> `l.p`

**Comment by kmaehashi** [comment]
LGTM!

**Comment by iwiwi** [comment]
Thank you so much!!

**Comment by kmaehashi** [comment]
For later reference, this fixes the issue (#4902) that ParallelUpdater does not share persistent values.
